src/js/map/point_data/filter_controller.js

```diff
@@ -145,6 +186,21 @@ export class FilterController extends Component {
 
         this.updateAvailableFiltersOfRows();
         this.model.dataFilterModel.updateFilteredData();
+
+        const addFilterRow = this.noFiltersAvailable && this.model.dataFilterModel.availableFilters.length > 0;   //no filters were available but now there are filterable fields
+        const removeFilterRow = !this.noFiltersAvailable && this.model.dataFilterModel.availableFilters.length <= 0;    //there were filterable fields but now there are none
+
+        this.noFiltersAvailable = this.model.dataFilterModel.availableFilters.length <= 0;
+
+        if (addFilterRow) {
```

Comment:
Making sure I understand correctly. So this.noFiltersAvailable is based on the current data has filters and the extra check is needed if the filters change when selecting another dataset?

Reply:
`this.noFiltersAvailable` is assigned `true` only when there are filterable fields